### PR TITLE
cobradocs sync

### PR DESCRIPTION
**Pull request by ajm188**
right now have this working through to syncing the docs for a _merged_ PR

i want to also provide support  for previewing doc changes based on either {main+the PR} or {releasebranch+the PR} if that nomenclature makes sense

finally i want to support also generate the sync pr when we _create_ tags for a release

### CODE REVIEW

File: go/cobradocs_sync.go
Comment (frouioui) [inline go/cobradocs_sync.go:64]: If i recall correctly, I think I had to amend the commit when doing a cherry-pick / create a new commit.

Comment (ajm188) [reply go/cobradocs_sync.go:80]: i meant this more in the sense that the `./tools/sync_cobradocs.sh` script in `website` authors a commit, and i'm not sure if, when running in prod, that will author as the bot user we want or not (nor am i sure how to test that without "just seeing what happens" haha). if it's authoring as the wrong user, then we'll need to amend that commit here to set the author/committer correctly

**Comment by ajm188** [comment]
@frouioui i'm going to split out that `CreateBranch` and `WithExtraEnv` changes to a separate PR because I think the release and preview changes are each going to be fairly large, so no sense holding up those fixes

**Comment by ajm188** [comment]
@frouioui i think we should punt on

> finally i want to support also generate the sync pr when we create tags for a release

to a separate PR so we can get this as well as #60 merged and i'll follow up with that feature next!

File: go/pull_request.go
Comment (frouioui) [inline go/pull_request.go:420]: let's add the number of the Pull Request in the debug message to ease debugging. this can be useful when you've got plenty of logs.

File: go/cobradocs_sync.go
Comment (frouioui) [inline go/cobradocs_sync.go:73]: should we remove this?

File: go/cobradocs_sync.go
Comment (frouioui) [reply go/cobradocs_sync.go:80]: Oh right. We can amend once we merge that in prod and see if it works :)

**Review by frouioui — APPROVED** [review]
This is really good thank you!

File: go/cobradocs_sync.go
Comment (ajm188) [reply go/cobradocs_sync.go:73]: yep!

File: go/pull_request.go
Comment (ajm188) [reply go/pull_request.go:420]: good call!

File: go/pull_request.go
Comment (ajm188) [inline go/pull_request.go:418]: forgot to come back to do this (hence the shared ListPRFiles function)

File: go/git/pull_request.go
Comment (frouioui) [inline go/git/pull_request.go:1]: ah yes this is much better, thanks!

**Review by frouioui — APPROVED** [review]

**Pull request merged**
